test/hero.test.js

```diff
@@ -0,0 +1,72 @@
+const expect = require('chai').expect;
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require('../server.js');
+const heroes = require('../state.js').heroes;
+chai.use(chaiHttp);
+
+describe('hero router', () => {
+    describe('GET /', () => {
+        it('should return heroes state', (done) => {
+            chai.request(app).get('/heroes').end((err, res) => {
+                expect(res.body[0].name).to.eq(heroes[0].name);
+                expect(res.body).to.have.lengthOf(2);
+                done();
+            });
+        });
+
+        it('should add a heroe to the state', (done) => {
+            chai.request(app).post('/heroes').send({ id: 3, name: 'Hero3' }).end((err, res) => {
+                expect(res.body[2].name).to.eq('Hero3');
+                expect(res.body).to.have.lengthOf(3);
+                done();
+            });
+        });
+
+        it('should delete a heroe to the state', (done) => {
+            chai.request(app).delete('/heroes?name=Hero3').end((err, res) => {
+                expect(res.body).to.have.lengthOf(2);
+                done();
+            });
+        });
+
+        it('should not find any hero to delete', (done) => {
+            chai.request(app).delete('/heroes?name=HeroUndefined').end((err, res) => {
+                expect(res.body).to.have.lengthOf(2);
```

Comment:
how do you know length of 2 is enough? should have checked the length before and after the chai request and then checked for a +-1 differnce